test/fn/compile-app-settings.spec.js

```diff
@@ -4,111 +4,152 @@ const sinon = require('sinon');
-
-  it('should handle a project with a purge function that need to be merged with other purge config', () =>
-    test('purge/merge-purging-function/project'));
-
-  it('should handle a project with no export purge config', () =>
-    test('purge/no-export-purge/project'));
-
-  it('should handle a project with correct purge config', () =>
-    test('purge/purge-correct/project'));
-
-  it('should reject a project where purge.fn is not a function', () =>
-    testFails('purge/purge-fn-not-a-function/project'));
-
-  it('should handle a project with a purge function', () =>
-    test('purge/purging-function/project'));
-
-  it('should reject a project with eslint error', () =>
-    testFails('eslint-error/project'));
-
-  it('can overwrite eslint rules with eslintrc file', () =>
-    test('eslintrc/project'));
-
-  it('should handle a configuration using the base_settings file', () =>
-    test('base-settings/project'));
-
-  it('should handle a configuration using the forms.json and schedules.json files', () =>
-    test('sms-modules/project'));
-
-  it('should reject a configuration using invalid forms.json or schedules.json files', () =>
-    testFails('sms-modules/invalid-files'));
-  
-  it('should reject a project with no .eslintrc file defined', () =>
-    testFails('missing-eslintrc/project'));
-});
-
-async function test(relativeProjectDir) {
-  const testDir = path.join(__dirname, '../data/compile-app-settings', relativeProjectDir);
-  sinon.stub(environment, 'pathToProject').get(() => testDir);
-  sinon.stub(environment, 'extraArgs').get(() => undefined);
-
-  // when
-  await compileAppSettings.execute();
-
-  // then
-  const actual = JSON.parse(JSON.stringify(writeJson.args[0][1]));
-  const expected = JSON.parse(fs.read(`${testDir}/../app_settings.expected.json`));
-  actual.tasks.rules = expected.tasks.rules = '';
-  actual.contact_summary = expected.contact_summary = '';
-  expect(actual).to.deep.eq(expected);
-}
-
-async function testFails(relativeProjectDir) {
-  const testDir = path.join(__dirname, '../data/compile-app-settings', relativeProjectDir);
-
-  // when
-  try {
-    await compileAppSettings.execute(testDir);
-    assert.fail('Expected assertion');
-  } catch (err) {
-    if (err.name === 'AssertionError') {
-      throw err;
-    }
-
-    assert.ok('asserted');
+  for (const scenario of scenarios) {
+    it(scenario.description, async () => {
+      const pathToTestProject = path.join(__dirname, '../data/compile-app-settings', scenario.folder);
+      sinon.stub(environment, 'pathToProject').get(() => pathToTestProject);
+      sinon.stub(environment, 'extraArgs').get(() => scenario.extraArgs);
+
+      const promiseToExecute = compileAppSettings.execute({ skipEslintIgnore: true });
+      if (scenario.error) {
+        try {
+          await promiseToExecute;
+          assert.fail('Expected execute() to throw');
+        } catch (err) {
+          expect(err.toString()).to.include(scenario.error);
+        }
```

Comment:
I find `chai-as-promised` to provide a really nice way to test promises that reject. It's not a dependency right now, but it's easy to add. With `chai-as-promised` these lines can be written as: 
```suggestion
        await expect(promiseToExecute).to.be.rejectedWith(Error, scenario.error);
```

Reply:
nice nice